### PR TITLE
Fixes issue #86

### PR DESCRIPTION
**Pull request by KikkiZ**
The document has been updated with a few additional changes to the links in the document.

In the process of updating, I accidentally synchronized my updates to the code to the repo, I have tried to roll back, not sure if it really solved the error.

PS: This update looks like a huge one, the documentation has changed a lot

### CODE REVIEW

File: doc/README-zhCN.md [hunk 1]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
+### 4 构建(Build)
+
+运行 `npm run build` 在生产模式下构建插件，然后用于安装的 xpi 和构建的代码在 `build` 文件夹中.
+
+`scripts/build.mjs` 的运行步骤:
+
+* 创建/清空 `build/`
+* 复制 `addon/**` 到 `build/addon/**`
+* 替换占位符：使用 `replace-in-file` 去替换在 `package.json` 中定义的关键字和配置 (`xhtml`、`.flt` 等)
+* 准备语言文件到 [avaid conflict](https://www.zotero.org/support/dev/zotero_7_for_developers#avoiding_localization_conflicts)
```
Comment (northword) [inline doc/README-zhCN.md:253]:
```suggestion
* 准备本地化文件以 [避免冲突](https://www.zotero.org/support/dev/zotero_7_for_developers#avoiding_localization_conflicts)
```

File: doc/README-zhCN.md [hunk 2]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
+```shell
+# A release-it command: version increase, npm run build, git push, and GitHub release
+# release-it: https://github.com/release-it/release-it
+npm run release
+```
+
+> [!note]
+> 在此模板中，release-it 被配置为在本地升级版本、构建、推送提交和 git 标签，随后GitHub Action 将重新构建插件并将 XPI 发布到 GitHub Release. 
+>
+> 如果你需要发布一个本地构建的 XPI，设置  `package.json` 中的 `release-it.github.release` 为 `true` 然后移除 `.github/workflows/release.yml`. 此外，你需要设置环境变量 `GITHUB_TOKEN`，它在[这里](https://github.com/settings/tokens)获取.
```
Comment (northword) [inline doc/README-zhCN.md:281]:
```suggestion
> 如果你需要发布一个本地构建的 XPI，将 `package.json` 中的 `release-it.github.release` 设置为 `true`，然后移除 `.github/workflows/release.yml`。此外，你需要设置环境变量 `GITHUB_TOKEN`，[获取 GitHub Token](https://github.com/settings/tokens)。
```

File: doc/README-zhCN.md [hunk 3]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
+
+#### 关于预发布
+
+该模板将 `prerelease` 定义为插件的测试版，当你在 release-it 中选择 `prerelease` 版本 (版本号中带有 `-` )，构建脚本将创建一个新的 `update-beta.json` 给预发布版本使用，这将确保常规版本的用户无法更新到测试版，只有手动下载并安装了测试版的用户才能自动更新到下一个测试版. 当下一个常规版本更新时，将更新 `update.json` 和 `update-beta.json`，以便常规版和测试版用户都可以更新到最新的常规版本.
```
Comment (northword) [inline doc/README-zhCN.md:285]:
```suggestion
该模板将 `prerelease` 定义为插件的测试版，当你在 release-it 中选择 `prerelease` 版本 (版本号中带有 `-` )，构建脚本将创建 `update-beta.json` 供预发布版本使用，这将确保正式版本的用户不会自动更新到测试版，只有手动下载并安装了测试版的用户才能自动更新到下一个测试版。当下一个正式版更新时，脚本将同时更新 `update.json` 和 `update-beta.json`，以便正式版和测试版用户都可以更新到最新的正式版。
```

File: doc/README-zhCN.md [hunk 4]
```diff
@@ -7,7 +7,22 @@
...
+
+[📖 Zotero 7 插件开发文档](https://www.zotero.org/support/dev/zotero_7_for_developers)
+
+🛠️ [Zotero 插件工具包](https://github.com/windingwind/zotero-plugin-toolkit) | [API 文档](https://github.com/windingwind/zotero-plugin-toolkit/blob/master/docs/zotero-plugin-toolkit.md)
+
+ℹ️ [Zotero 类型定义](https://github.com/windingwind/zotero-types)
+
+📜 [Zotero 源代码](https://github.com/zotero/zotero)
+
+📌 [Zotero 插件模板](https://github.com/windingwind/zotero-plugin-template) (即当前库)
```
Comment (northword) [inline doc/README-zhCN.md:20]:
```suggestion
📌 [Zotero 插件模板](https://github.com/windingwind/zotero-plugin-template) (即本仓库)
```

File: doc/README-zhCN.md [hunk 5]
```diff
@@ -7,7 +7,22 @@
...
+
+> [!tip]
+> 👁 关注此库，以便在有修复或更新时及时收到通知.
```
Comment (northword) [inline doc/README-zhCN.md:23]:
```suggestion
> 👁 Watch 本仓库，以及时收到修复或更新的通知.
```

File: doc/README-zhCN.md [hunk 6]
```diff
@@ -50,19 +52,21 @@
...
-- 插件构建/测试/发布工作流:
-  - 自动生成/更新插件id和版本、更新配置和设置环境变量 (`development/production`)；
+- 插件开发/构建/发布工作流:
+  - 自动生成/更新插件id和版本、更新配置和设置环境变量 (`development`/`production`)；
   - 自动在 Zotero 中构建和重新加载代码；
   - 自动发布到GitHub (使用[release-it](https://github.com/release-it/release-it));
+- 集成Prettier和ES Lint;
+
+> [!warning]
+> Zotero系统已升级(dtd 已弃用，我们将不在使用 .properties). 主分支将只支持 Zotero 7.0.0-beta.12 或更高版本. 如果需要支持 Zotero 6，可能需要同时使用`dtd`、`properties` 和`ftl`. 请参考此库的 `zotero6-bootstrap` 分支.
```
Comment (northword) [inline doc/README-zhCN.md:69]:
```suggestion
> Zotero 本地化系统已升级（`dtd` 已弃用，我们将不在使用 `.properties`)。 主分支目前只支持 Zotero 7.0.0-beta.12 及更高版本。如果需要支持 Zotero 6，你可能需要同时使用 `dtd`、`properties` 和`ftl`， 请参考 `zotero6-bootstrap` 分支。
```

File: doc/README-zhCN.md [hunk 7]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
 
 ## Quick Start Guide 快速入门指南
 
-### 安装预构建 `xpi`
+### 0 环境要求(Requirement)
```
Comment (northword) [inline doc/README-zhCN.md:136]:
```suggestion
### 0 前置要求
```

File: doc/README-zhCN.md [hunk 8]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
 
-通过直接在GitHub中下载构建好的 `xpi` 文件并将其安装到Zotero中来了解示例的工作原理.
+1. 安装beta版Zotero: https://www.zotero.org/support/beta_builds
```
Comment (northword) [inline doc/README-zhCN.md:138]:
```suggestion
1. 安装测试版 Zotero: https://www.zotero.org/support/beta_builds
```

File: doc/README-zhCN.md [hunk 9]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
+2. 安装 [Node.js](https://nodejs.org/en/) 和 [Git](https://git-scm.com/)
 
-这也是你发布插件的格式，同时这也将是其他人可以直接使用的版本.
+> [!note]
+> 本指南默认你已经对Zotero插件的基础结构和工作原理有初步的了解. 如果你还不了解，请先参考 [文档](https://www.zotero.org/support/dev/zotero_7_for_developers) 和官方插件样例 [Make It Red](https://github.com/zotero/make-it-red) .
```
Comment (northword) [inline doc/README-zhCN.md:142]:
```suggestion
> 本指南默认你已经对 Zotero 插件的基本结构和工作原理有初步的了解. 如果你还不了解，请先参考 [文档](https://www.zotero.org/support/dev/zotero_7_for_developers) 和官方插件样例 [Make It Red](https://github.com/zotero/make-it-red) 。
```

File: doc/README-zhCN.md [hunk 10]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
-      addonRef, // e.g. Element ID prefix
-      addonInstance // the plugin's root instance: Zotero.${addonInstance}
-    }
-  }
-  ```
+### 1 创建你的仓库(Create Your Repo)
 
-  > 注意设置 addonID 和 addonRef 以避免冲突.
+1. 点击 `Use this template`；
+2. 使用 `git clone` 克隆此库；
```
Comment (northword) [inline doc/README-zhCN.md:147]:
```suggestion
2. 使用 `git clone` 克隆前一步生成的仓库；
```

clone的是生成的仓库，不是本仓库

File: doc/README-zhCN.md [hunk 11]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
-如果要构建和发布插件，运行如下指令：
+3. 进入项目文件夹；
 
-```shell
-# A release-it command: version increase, npm run build, git push, and GitHub release
-# You need to set the environment variable GITHUB_TOKEN https://github.com/settings/tokens
-# release-it: https://github.com/release-it/release-it
-npm run release
-```
+### 2 调整模板设置和环境(Config Template Settings and Enviroment)
```
Comment (northword) [inline doc/README-zhCN.md:161]:
```suggestion
### 2 配置模板和开发环境
```

File: doc/README-zhCN.md [hunk 12]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
-   > (可选项) 此操作仅需执行一次: 使用 `/path/to/zotero -p` 启动 Zotero，创建一个新的配置文件并用作开发配置文件.
-   > 将配置文件的路径 `profilePath` 放入 `zotero-cmd.json` 中，以指定要使用的配置文件.
+    > [!warning]
+    > 注意设置 addonID 和 addonRef 以避免冲突.
 
-   ```sh
-   cp ./scripts/zotero-cmd-default.json ./scripts/zotero-cmd.json
-   vim ./scripts/zotero-cmd.json
-   ```
+如果你需要在GitHub意外的地方托管你的 XPI 包，请删除 `releasePage` 并添加 `updateLink`，并将值设置为你的 XPI 下载地址.
```
Comment (northword) [inline doc/README-zhCN.md:186]:
```suggestion
如果你需要在 GitHub 以外的地方托管你的 XPI 包，请删除 `releasePage`，添加 `updateLink`，并将值设置为你的 XPI 下载地址。
```

File: doc/README-zhCN.md [hunk 13]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
 
-4. 构建插件并使用 `npm run restart` 重启 Zotero.
+2. 复制 Zotero 命令行配置文件. 修改启动你之前安装的 beta 版 Zotero 的命令
```
Comment (northword) [inline doc/README-zhCN.md:188]:
```suggestion
2. 复制 Zotero 启动配置，填入 Zotero 可执行文件路径和 profile 路径
```

File: README.md [hunk 14]
```diff
@@ -278,14 +278,14 @@ npm run release
 > [!note]
 > In this template, release-it is configured to locally bump the version, build, and push commits and git.tags, subsequently GitHub Action will rebuild the plugin and publish the XPI to GitHub Release.
 >
-> If you need to release a locally built XPI, set `release-it.github.release` to `true` in `package.json` and remove `.github/workflows/release.yml`. Besides that, you need to set the environment variable `GITHUB_TOKEN`, get it in <https://github.com/settings/tokens>
+> If you need to release a locally built XPI, set `release-it.github.release` to `true` in `package.json` and remove `.github/workflows/release.yml`. Besides that, you need to set the environment variable `GITHUB_TOKEN`, get it in [here](https://github.com/settings/tokens).
```
Comment (windingwind) [inline README.md:281]: I suppose @northword put these links' actual url here intentionally?

File: doc/README-zhCN.md [hunk 15]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
 
-5. 启动 Firefox 102 (Zotero 7)
+	> (可选项) 此操作仅需执行一次: 使用 `/path/to/zotero -p` 启动 Zotero，创建一个新的配置文件并用作开发配置文件.
+	> 将配置文件的路径 `profilePath` 放入 `zotero-cmd.json` 中，以指定要使用的配置文件.
 
-6. 在 Firefox 中，转到devtools，转到设置，单击 "enable remote debugging" ，同时，旁边的按钮也是关于调试的。
+	```sh
+	cp ./scripts/zotero-cmd-default.json ./scripts/zotero-cmd.json
```
Comment (northword) [inline doc/README-zhCN.md:194]:
```suggestion
	cp ./scripts/zotero-cmd-template.json ./scripts/zotero-cmd.json
```

File: doc/README-zhCN.md [hunk 16]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
+	vim ./scripts/zotero-cmd.json
+	```
 
-   > 在 FirFox 102 中输入 `about:debugging#/setup` .
+3. 运行 `npm install` 以安装相关依赖
 
-7. 在 Zotero 中，进入设置-高级-编辑器，搜索 "debugging" 然后单击 "allow remote debugging".
+	> 如果你使用 `pnpm` 作为你项目的包管理器，你需要添加 `public-hoist-pattern[]=*@types/bluebird*` 到`.npmrc`, 详情请[跳转这里](https://github.com/windingwind/zotero-types?tab=readme-ov-file#usage) .
```
Comment (northword) [inline doc/README-zhCN.md:200]:
```suggestion
	> 如果你使用 `pnpm` 作为包管理器，你需要添加 `public-hoist-pattern[]=*@types/bluebird*` 到`.npmrc`, 详情请参阅 [zotero-types](https://github.com/windingwind/zotero-types?tab=readme-ov-file#usage) .
```

File: doc/README-zhCN.md [hunk 17]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
 
-8. 在 Firefox 中连接 Zotero. 在 FireFox 102中，在远程调试页面底部输入 `localhost:6100` 然后单击 `add`.
+### 3 编码(Coding)
```
Comment (northword) [inline doc/README-zhCN.md:202]:
```suggestion
### 3 开始开发
```

File: doc/README-zhCN.md [hunk 18]
```diff
@@ -272,7 +305,7 @@ npm run release
 
 ### 关于全局变量(About Global Variables)
 
-> 可以在 [`src/index.ts`](https://github.com/windingwind/zotero-plugin-template/blob/bootstrap/src/index.ts) 中查看更多
+> 可以在 [`src/index.ts`](https://github.com/windingwind/zotero-plugin-template/blob/main/src/index.ts)中查看更多
 
 引导插件在沙盒中运行，但沙盒中没有默认的全局变量，例如 `Zotero` 或 `window` 等我们曾在覆盖插件环境中使用的变量.
```
Comment (windingwind) [inline doc/README-zhCN.md:310]: just noticed, `引导插件` should be `bootstrap插件` and `覆盖插件` be `overlay插件`

File: doc/README-zhCN.md [hunk 19]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
 
-8. 在 Firefox 中连接 Zotero. 在 FireFox 102中，在远程调试页面底部输入 `localhost:6100` 然后单击 `add`.
+### 3 编码(Coding)
 
-9. 在远程调试页面左侧栏点击 `connect`.
+使用 `npm start` 启动开发服务器，它将：
 
-10. 点击 "Inspect Main Process"
+* 在开发模式下预构建插件
+* 让 Zotero 从`build/` 中加载插件
```
Comment (northword) [inline doc/README-zhCN.md:207]:
```suggestion
* 启动 Zotero，令其从 `build/` 中加载插件
```

File: doc/README-zhCN.md [hunk 20]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
+* 开启开发工具（devtool）
```
Comment (northword) [inline doc/README-zhCN.md:208]:
```suggestion
* 打开开发者工具（devtool） 
```

File: doc/README-zhCN.md [hunk 21]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
+### 3 编码(Coding)
 
-9. 在远程调试页面左侧栏点击 `connect`.
+使用 `npm start` 启动开发服务器，它将：
 
-10. 点击 "Inspect Main Process"
+* 在开发模式下预构建插件
+* 让 Zotero 从`build/` 中加载插件
+* 开启开发工具（devtool） 
+* 注意 `src/**` 和 `addon/**`.
```
Comment (northword) [inline doc/README-zhCN.md:209]:
```suggestion
* 监听 `src/**` 和 `addon/**`.
```

**Review by windingwind — CHANGES_REQUESTED** [review]

File: doc/README-zhCN.md [hunk 22]
```diff
@@ -129,136 +133,165 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
 - 使用 `Zotero.debug()` 调试输出. 在 Help->Debug Output Logging->View Output 查看输出;
+
 - 调试 UI. Zotero 建立在 Firefox XUL 框架之上. 使用 [XUL Explorer](https://udn.realityripple.com/docs/Archive/Mozilla/XUL_Explorer) 等软件调试 XUL UI.
-  > XUL 文档: <http://www.devdoc.net/web/developer.mozilla.org/en-US/docs/XUL.html>
+
+    > XUL 文档: <http://www.devdoc.net/web/developer.mozilla.org/en-US/docs/XUL.html>
+
+### 4 构建(Build)
+
+运行 `npm run build` 在生产模式下构建插件，然后用于安装的 xpi 和构建的代码在 `build` 文件夹中.
```
Comment (northword) [inline doc/README-zhCN.md:246]:
```suggestion
运行 `npm run build` 在生产模式下构建插件，用于安装的 xpi 和构建的代码在 `build` 文件夹中.
```

File: README.md (hunk 14 above)
Comment (northword) [reply README.md:281]: 这个倒是不影响，虽然我确实是故意直接放网址的 :>

Comment (northword) [reply README.md:281]: 这个倒是不影响，虽然我确实是故意直接放网址的 :>

File: doc/README-zhCN.md (hunk 6 above)
Comment (windingwind) [reply doc/README-zhCN.md:69]: 不在 -> 不再

**Comment by northword** [comment]
> In the process of updating, I accidentally synchronized my updates to the code to the repo, I have tried to roll back, not sure if it really solved the error.

这个包含以往的提交，是因为上一次 PR 是以 squash 的方式合并进仓库的，这意味着 PR 发起（from）分支的所有提交将被压缩为一个提交进入本仓库 main 分支；而 sync to upstream 是创建一个 merge 提交到 fork 仓库，这使得 upstream/main 与 fork/main 无法保持一致。

当向其他仓库 PR 时，最好在新的分支中（从upstream/main 创建分支）提交内容然后发起 PR。

现在的 main 可以在本地强制推送以和本仓库的提交记录保持一致。

File: README.md (hunk 14 above)
Comment (KikkiZ) [reply README.md:281]: okok，我之后会注意修改链接的内容，尽量避免 `here` 这类标签

File: doc/README-zhCN.md (hunk 21 above)
Comment (KikkiZ) [reply doc/README-zhCN.md:209]: 我建议将原文中的 `watch` 改为 `listening` 可能会更加易于理解

Comment (northword) [reply doc/README-zhCN.md:209]: > 我建议将原文中的 `watch` 改为 `listening` 可能会更加易于理解

watch 和 listening 是两个术语，这里是 watch 没有问题。中文也可以选择翻译为 “监视” 以示区别。

File: README.md [hunk 23]
```diff
@@ -56,7 +56,7 @@ If you are using this repo, I recommended that you put the following badge on yo
 
 - Event-driven, functional programming, under extensive skeleton;
 - Simple and user-friendly, works out-of-the-box.
-- ⭐ [New!] Auto hot reload! Whenever the source code is modified, automatically compile and reload. [See here→](#auto-hot-reload)
+- ⭐ [New!] Auto hot reload! Whenever the source code is modified, automatically compile and reload. See more in [section auto hot reload](#auto-hot-reload)
```
Comment (windingwind) [inline README.md:59]: [section auto hot reload] seems not to be a meaningful sentence

File: README.md [hunk 24]
```diff
@@ -330,7 +330,7 @@ createElement(document, "button", { namespace: "xul" }); // manually set namespa
 
 ### About Zotero API
 
-Zotero docs are outdated and incomplete. Clone <https://github.com/zotero/zotero> and search the keyword globally.
+Zotero docs are outdated and incomplete. Clone [Zotero Code Repo](https://github.com/zotero/zotero) and search the keyword globally.
```
Comment (windingwind) [inline README.md:333]: we'd better keep the raw link here so that people can directly copy the link and clone

File: doc/README-zhCN.md [hunk 25]
```diff
@@ -129,136 +133,166 @@ Obsidian风格的指令输入模块，它通过接受文本来运行插件，并
...
 当检测到 `src` 或 `addon` 中的文件修改时，插件将自动编译并重新加载.
 
 <details style="text-indent: 2em">
 <summary>💡 将此功能添加到现有插件的步骤</summary>
 
-1. 复制 `scripts/reload.mjs`
-2. 复制 `reload` 、`watch` 和 `start-watch` 命令 `package.json`
-3. 运行 `npm install --save-dev chokidar-cli`
+1. 复制 `scripts/**.mjs`
+2. 复制 `server` 、`build` 和 `stop` 命令 `package.json`
```
Comment (windingwind) [inline doc/README-zhCN.md:227]: 复制...到...

File: src/modules/examples.ts [hunk 26]
```diff
@@ -191,9 +191,8 @@ export class UIExampleFactory {
       },
     });
     document.documentElement.appendChild(styles);
-    document
-      .getElementById("zotero-item-pane-content")
-      ?.classList.add("makeItRed");
+    document.getElementById("zotero-item-pane-content")
+      ?.classList.add("make");
```
Comment (windingwind) [inline src/modules/examples.ts:195]: please revert these changes

**Review by windingwind — CHANGES_REQUESTED** [review]

File: README.md (hunk 24 above)
Comment (KikkiZ) [reply README.md:333]: Ok, I'll roll back these changes and take care of these issues later.

**Comment by KikkiZ** [comment]
I rolled back all the links in the English document to their original state, and then I only modified a few punctuation marks in the document, and added a git homepage link.
I also changed the hidden links in the Chinese document to be directly displayed, but I still added necessary descriptions before the links so that users can understand the function of the links. If you think this supplement is unnecessary, I can change it back to its original appearance.

File: src/modules/examples.ts (hunk 26 above)
Comment (windingwind) [reply src/modules/examples.ts]: I mean, the changes to the `examples.ts` should be reverted.

Comment (KikkiZ) [reply src/modules/examples.ts]: I previously rolled back the code, and I don't know why I synchronized the wrong code again. I am really sorry.

**Comment by windingwind** [comment]
thanks!
